syntax = "proto3";
package net;
message NetMessage{
  oneof body{
    GameServerConfig server_config = 5;
    ClientInfo client_info = 6;
    GenerateRoom generate_room_request = 7;
    ConnectRoom connect_room = 8;
    GameMessage game_message = 1;
    ChatMessage chat_message = 2;
  }
}

message GameServerConfig{
  oneof server_address{
    fixed32 v4 = 1;
    bytes v6 = 2;
  };
  uint32 server_state_port = 3;
  uint32 server_command_port = 4;
  string room_code = 5;
}

message ClientInfo{
  string client_name = 1;
}

message GenerateRoom{
}

message ConnectRoom{
  string room_code = 4;
}

message GameMessage{
  string message = 1; // plug (however)
}

message ChatMessage{
  string message = 1;
}